pkg/kvstore/lock.go

```diff
@@ -67,9 +69,16 @@ func LockPath(path string) (*Lock, error) {
 
 // Unlock unlocks a lock
 func (l *Lock) Unlock() error {
+	// Unlock kvstore mutex first
 	err := l.lock.Unlock()
 
-	lockPaths[l.path].Unlock()
+	lockPathsMU.RLock()
+	localLock, ok := lockPaths[l.path]
```

Comment:
Why is this change needed?

Reply:
We are reading from `lockPaths` so we need to have a read lock to protect that "read"

Reply:
Why do we need to re-lookup the lock variable?

Reply:
Discussed offline, the changes allow to cleanup the map

Reply:
Discussed offline, the changes allow to cleanup the map